plugin/trino-pulsar/src/main/java/io/trino/plugin/pulsar/PulsarSplitManager.java

```diff
@@ -0,0 +1,500 @@
+import static org.apache.bookkeeper.mledger.ManagedCursor.FindPositionConstraint.SearchAllAvailableEntries;
+
+/**
+ * The class helping to manage Trino Pulsar splits.
+ */
+public class PulsarSplitManager
+        implements ConnectorSplitManager
+{
+    private static final Logger log = Logger.get(PulsarSplitManager.class);
+
+    private final String catalogName;
+
+    private final PulsarConnectorConfig pulsarConnectorConfig;
+
+    private final PulsarConnectorCache pulsarConnectorManagedLedgerFactory;
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Inject
+    public PulsarSplitManager(
+            CatalogName catalogName,
+            PulsarConnectorConfig pulsarConnectorConfig,
+            PulsarConnectorCache pulsarConnectorManagedLedgerFactory)
+    {
+        this.catalogName = requireNonNull(catalogName, "catalogName is null").toString();
+        this.pulsarConnectorConfig = requireNonNull(pulsarConnectorConfig, "pulsarConnectorConfig is null");
+        this.pulsarConnectorManagedLedgerFactory = requireNonNull(pulsarConnectorManagedLedgerFactory, "pulsarConnectorManagedLedgerFactory is null");
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transactionHandle,
+            ConnectorSession session,
+            ConnectorTableLayoutHandle layout,
+            SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        int numSplits = this.pulsarConnectorConfig.getTargetNumSplits();
+
+        PulsarTableLayoutHandle layoutHandle = (PulsarTableLayoutHandle) layout;
+        PulsarTableHandle tableHandle = layoutHandle.getTable();
+        TupleDomain<ColumnHandle> tupleDomain = layoutHandle.getTupleDomain();
+
+        String namespace = PulsarConnectorUtils.restoreNamespaceDelimiterIfNeeded(tableHandle.getSchemaName(), pulsarConnectorConfig);
+        TopicName topicName = TopicName.get("persistent", NamespaceName.get(namespace), tableHandle.getTopicName());
+
+        SchemaInfo schemaInfo;
+
+        try (PulsarAdmin pulsarAdmin = PulsarAdminClientProvider.getPulsarAdmin(pulsarConnectorConfig)) {
+            schemaInfo = pulsarAdmin.schemas().getSchemaInfo(format("%s/%s", namespace, tableHandle.getTopicName()));
+        }
+        catch (PulsarAdminException e) {
+            if (e.getStatusCode() == 401) {
+                throw new TrinoException(QUERY_REJECTED, format("fail to get pulsar topic schema for topic %s/%s: Unauthorized", namespace, tableHandle.getTopicName()));
+            }
+            else if (e.getStatusCode() == 404) {
+                schemaInfo = PulsarSchemaInfoProvider.defaultSchema();
+            }
+            else {
+                throw new TrinoException(PULSAR_SPLIT_ERROR, "fail to get pulsar topic schema", e);
+            }
+        }
+        catch (PulsarClientException e) {
+            throw new TrinoException(PULSAR_ADMIN_ERROR, "fail to create pulsar admin client", e);
```

Comment:
TrinoException is user facing error. Please work  a bit more on error message so user can actually understand what happened. 

What is `PULSAR_ADMIN_ERROR`? When user gets this what they should do? Same for other error codes and please double check all places where you throw TrinoException